Buildings/Controls/OBC/ASHRAE/G36_PR1/Generic/SetPoints/OperationMode.mo

```diff
@@ -224,14 +218,26 @@ protected
   Buildings.Controls.OBC.CDL.Conversions.RealToInteger setUpMod
     "Convert Real number to Integer number"
     annotation (Placement(transformation(extent={{200,-220},{220,-200}})));
-  Buildings.Controls.OBC.CDL.Conversions.BooleanToInteger booToInt1(
-    final integerTrue=Buildings.Controls.OBC.ASHRAE.G36_PR1.Types.OperationModes.warmUp)
+  Buildings.Controls.OBC.CDL.Logical.Edge edg
```

Comment:
If using `egde`here, then how to handle this case:

- when at the moment when it should be in the `cool-down` model according to the `maxCooDowTim`, so that `edg` will output a `true` and then it becomes `u1` input to `and2`.
- but at the same moment, the `uHigOccCoo` is still `false` (all the zone temperature are still below the occupied cooling setpoint), so the `u2` input of `and2` will be `false`.
- then it will never go into the cooling-down mode since the `and2` will never give a `true` output.  

The `and2` should be able to give a `true` output: 
- during the cool-down period time,
- anyone zone temperature becomes higher than its occupied cooling setpoint (`uHigOccCoo` becomes true -- maximum zone temperature in the same AHU group becomes higher than occupied cooling setpoint).

Similar scenario should also be considered for the case of warm-up mode.

Reply:
You are right. I did not consider this case. I re-implemented the logic. I will push the updates.